synapse/cores/common.py

```diff
@@ -259,67 +297,54 @@ def setModlVers(self, name, vers):
             self.setRowsByIdProp(iden, prop, vers)
             return vers
 
-    def revModlVers(self, name, revs):
-        '''
-        Update a model using a list of (vers,func) tuples.
+    def _addDataModels(self, modtups):
 
-        Args:
-            name (str): The name of the model
-            revs ([(int,function)]):  List of (vers,func) revision tuples.
+        DataModel._addDataModels(self, modtups)
 
-        Returns:
-            (None)
-
-        Example:
+        for name, modl in modtups:
 
-            with s_cortex.openurl('ram:///') as core:
+            for tnam, tnfo in modl.get('types', ()):
+                self.addRuntNode('syn:type', tnam, **tnfo)
 
-                def v0():
-                    addModelStuff(core)
+            for fnam, fnfo, props in modl.get('forms', ()):
 
-                def v1():
-                    addMoarStuff(core)
+                self.addRuntNode('syn:form', fnam, **fnfo)
+                self.addRuntNode('syn:prop', fnam, **fnfo)
 
-                revs = [ (0,v0), (1,v1) ]
+                for pnam, pnfo in props:
+                    full = fnam + ':' + pnam
+                    self.addRuntNode('syn:prop', full, **pnfo)
 
-                core.revModlVers('foo', revs)
-
-        Each specified function is expected to update the cortex including data migration.
+    def revModlVers(self, name, vers, func):
         '''
-        if not revs:
-            return
-
-        curv = self.getModlVers(name)
+        Update and track a model version using a callback function.
 
-        maxver = revs[-1][0]
-        if maxver == curv:
-            return
+        Args:
+            name (str): The name of the model
+            vers (int): The version int ( in YYYYMMDDHHMM int format )
```

Comment:
with the removal of revcoremodl from CoreModule class we no longer have any validation being done the vers string is in the format.

Reply:
but vers isn't a string...  not sure i follow.

Reply:
rather - the version integer isn't in that format.